src/main/java/no/sikt/nva/BrageMigrationCommand.java

```diff
@@ -124,12 +128,37 @@ public BrageMigrationCommand() {
         this.s3Client = S3Driver.defaultS3Client().build();
     }
 
+    private static void validateAwsEnvironment(String environment) {
+        var accountId = getAccountId();
+        var awsEnvironment = AwsEnvironment.fromValue(environment);
+        if (doesNotBelongToCurrentAwsEnvironment(accountId, awsEnvironment)) {
+           throw new IllegalArgumentException(WRONG_ACCOUNT_MESSAGE + AwsEnvironment.getEnvironmentById(accountId).getValue());
+        }
+    }
+
+    private static boolean doesNotBelongToCurrentAwsEnvironment(String accountId, AwsEnvironment awsEnvironment) {
+        return nonNull(accountId)
+               && nonNull(awsEnvironment.getAccountId())
+               && !accountId.equals(awsEnvironment.getAccountId());
+    }
+
+    private static String getAccountId() {
+        return AWSSecurityTokenServiceClientBuilder
+                   .standard()
+                   .withRegion(Region.EU_WEST_1.id())
+                   .build()
+                   .getCallerIdentity(new GetCallerIdentityRequest())
+                   .getAccount();
+    }
+
     public BrageMigrationCommand(S3Client s3Client) {
         this.s3Client = s3Client;
     }
 
     public static void main(String[] args) {
         setSystemPropertiesForLogFiles(args);
+        var arg = getArgument(Arrays.asList(args), "-j", "--aws-bucket");
+        validateAwsEnvironment(arg.orElse(null));
```

Comment:
Adding it here to validate only when we running "prod" constructor, can alternatively pass this argument directly to prod constructor.